File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
  const set = new Set();
  for (const v of phone_book) {
    set.add(v);
  }
    
  for (const v of phone_book) {
    for (let i = 1; i < v.length; i++) {
      const prefix = v.slice(0, i);

      if (set.has(prefix)) {
        return false;
      }
    }
  }
    
  return true;
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
# 1,000,000 -> 이중 루프 사용 불가
def solution(phone_book):
    # default result value
    answer = True
    
    # get value : O(1)
    tb = set(phone_book)
    
    for item in tb:
        for idx in range(len(item)):
            prefix = item[:idx]
            if prefix in tb:
                answer = False
                break
    
    return answer